mirgecom/artificial_viscosity.py

```diff
@@ -200,20 +204,25 @@ def av_laplacian_operator(discr, boundaries, fluid_state, alpha, gas_model=None,
     quadrature_tag
         An optional identifier denoting a particular quadrature
         discretization to use during operator evaluations.
-        The default value is *None*.
 
-    boundary_kwargs: :class:`dict`
-        dictionary of extra arguments to pass through to the boundary conditions
+    volume_dd: grudge.dof_desc.DOFDesc
+        The DOF descriptor of the volume on which to apply the operator.
 
     Returns
     -------
     :class:`mirgecom.fluid.ConservedVars`
         The artificial viscosity operator applied to *q*.
     """
+    boundaries = {
+        as_dofdesc(bdtag).domain_tag: bdry
+        for bdtag, bdry in boundaries.items()}
+
     cv = fluid_state.cv
     actx = cv.array_context
-    dd_vol = DOFDesc("vol", quadrature_tag)
-    dd_faces = DOFDesc("all_faces", quadrature_tag)
+
+    dd_base = volume_dd
+    dd_vol = dd_base.with_discr_tag(quadrature_tag)
```

Comment:
I'm getting myself confused on how `dd_base`, `dd_vol`, `volume_dd` are intended to be different. (Add a comment?)

Reply:
I did my best to clarify the naming. I'm now using `dd` for the function parameters, checking that they're volumes where required, and then using `dd_vol` and `dd_vol_quad` internally for the base and quadrature discretizations, respectfully (and using the same naming scheme for other `dd`s like `bdry`/`allfaces` and arrays).